extension/basicauth/basicauth.go

```diff
@@ -0,0 +1,140 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package basicauth // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/basicauth"
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/tg123/go-htpasswd"
+	"go.opentelemetry.io/collector/client"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configauth"
+)
+
+var _ configauth.ServerAuthenticator = (*BasicAuth)(nil)
+
+type BasicAuth struct {
+	htpasswd  HtpasswdSettings
+	matchFunc matchFunc
+}
+
+type matchFunc func(username, password string) bool
+
+func (ba *BasicAuth) Start(ctx context.Context, host component.Host) error {
```

Comment:
Please, use the helper to create the new extension:  `configauth.NewServerAuthenticator`. Like this:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a395bdfd5f5145672a2f8c73a88f94a15b516c3f/extension/oidcauthextension/extension.go#L70-L75